### ISSUE TITLE
Consider adding timeout value to script execution

### ISSUE THREAD

**Issue by nafai** [Ylianst/MeshCentral#6076]
As seen in #2202, presenting in the agent console message _Run commands can't execute, already busy._

By running a script that pauses for any reason (waiting on stdin, waiting on an interactive window, etc), an agent can get "stuck" and not process any other scripts you try to send it. An agent will remain in this state until the service is restarted, running `agentupdate`, etc.

Please consider adding a timeout value for script execution in config.json. Even a long timeout (1800/3600 secs as a default value) would be preferable to an indefinite timeout.

Alternatives could include adding the timeout as another field in the Run Commands dialog, or to just launch multiple scripts in parallel and not care if they ever finish executing.

**Comment by DaanSelen** [comment]
I can also say that this happens using `MeshCtrl.js` its way of interacting with MeshCentral, I assume this is the same thing as the Run Command dialog box.

**Comment by si458** [comment]
Can anyone provide me with an example command u run that takes a long time and never times out but should?

**Comment by DaanSelen** [comment]
> Can anyone provide me with an example command u run that takes a long time and never times out but should?

My example was with an `apt` command on Linux without a certain `-y` flag. This makes `apt` wait for stdin like: `Do you accept? [Y/n]`

This caused my agent to lock up.

**Comment by si458** [comment]
@DaanSelen oh yeh great thanks! Never thought of that yet I use that command manually every week for server updates haha, one of those weeks!

**Comment by DaanSelen** [comment]
> @DaanSelen oh yeh great thanks! Never thought of that yet I use that command manually every week for server updates haha, one of those weeks!

No worries! But because this command is SOOO essential I thought to make a report of this.

**Comment by DaanSelen** [comment]
Hello Simon, is this still an issue?

**Comment by si458** [comment]
yes sorry i havent had change to look into this and add the feature, ill try have another look next week!

**Comment by DaanSelen** [comment]
Bump @si458 following the MeshCentral meeting.

**Comment by Tech-Werkz** [comment]
is there an update regarding this issue  🙂 ?

**Comment by DaanSelen** [comment]
> is there an update regarding this issue 🙂 ?

Not yet, if you are able. Also participate in the meeting to show animo!

**Comment by DaanSelen** [comment]
Bump

**Comment by Tech-Werkz** [comment]
im not sure it its the right fix but i saw a command on console tab named "tasklimiter"
i tried searching thru web and manuals but i did not see any meaning, upon checking on vscode  its pointing to the console logs result( just a guess) maybe it can help because any replies from a command, can be viewed on console
or maybe can someone explain the use of the tasklimiter command on server console tab

**Comment by DaanSelen** [comment]
THIS IS NOT STALE.

**Comment by DaanSelen** [comment]
Bump

**Comment by DaanSelen** [comment]
Not stale

**Comment by nafai** [comment]
Not stale

**Comment by DaanSelen** [comment]
Not Stale

**Comment by DaanSelen** [comment]
Not stale I think

**Comment by nafai** [comment]
Nope, not stale.
